### PR TITLE
Improve documentation for the TLS  section of the provider connection.

### PR DESCRIPTION
**Pull request by ldez**
### What does this PR do?

Improve documentation for the TLS  section of the provider connection.

### Motivation

Remove TODO

### More

- [ ] ~~Added/updated tests~~
- [x] Added/updated documentation

### CODE REVIEW

File: docs/content/middlewares/forwardauth.md [hunk 1]
```diff
@@ -206,7 +206,7 @@ The `tls` option is the TLS configuration from Traefik to the authentication ser
 
 #### `tls.ca`
 
-TODO add description.
+Certificate Authority used for the connection to the authentication server.
```
Comment (dduportal) [inline docs/content/middlewares/forwardauth.md]:
```suggestion
Certificate Authority used for the secured connection to the authentication server.
```

File: docs/content/middlewares/forwardauth.md [hunk 2]
```diff
@@ -266,7 +266,12 @@ http:
 
 #### `tls.caOptional`
 
-TODO add description.
+The policy followed by the server for TLS Client Authentication.
+Only used if `tls.ca` is defined.
```
Comment (dduportal) [inline docs/content/middlewares/forwardauth.md]:
```suggestion
Requires `tls.ca` to be defined.
```

File: docs/content/middlewares/forwardauth.md [hunk 3]
```diff
@@ -266,7 +266,12 @@ http:
 
 #### `tls.caOptional`
 
-TODO add description.
+The policy followed by the server for TLS Client Authentication.
```
Comment (dduportal) [inline docs/content/middlewares/forwardauth.md]:
```suggestion
Policy used for the secured connection with TLS Client Authentication to the authentication server.
```

File: docs/content/middlewares/forwardauth.md [hunk 4]
```diff
@@ -316,7 +321,7 @@ http:
 
 #### `tls.cert`
 
-TODO add description.
+Public certificate used for the connection to the authentication server.
```
Comment (dduportal) [inline docs/content/middlewares/forwardauth.md]:
```suggestion
Public certificate used for the secured connection to the authentication server.
```

File: docs/content/middlewares/forwardauth.md [hunk 5]
```diff
@@ -385,7 +390,7 @@ http:
 
 #### `tls.key`
 
-TODO add description.
+Private certificate used for the connection to Docker.
```
Comment (dduportal) [inline docs/content/middlewares/forwardauth.md]:
```suggestion
Private certificate used for the secure connection to the authentication server.
```

File: docs/content/middlewares/forwardauth.md (hunk 3 above)
Comment (dduportal) [reply docs/content/middlewares/forwardauth.md]: - Stay coherent with the other sentences (removing the `The` at the beginning)
- Clarify what "server" is

File: docs/content/middlewares/forwardauth.md [hunk 6]
```diff
@@ -266,7 +266,12 @@ http:
 
 #### `tls.caOptional`
 
-TODO add description.
+The policy followed by the server for TLS Client Authentication.
+Only used if `tls.ca` is defined.
+
+- `true`: VerifyClientCertIfGiven
```
Comment (dduportal) [inline docs/content/middlewares/forwardauth.md]: Do we have a reference to what `VerifyClientCertIfGiven` is?

File: docs/content/middlewares/forwardauth.md [hunk 7]
```diff
@@ -266,7 +266,12 @@ http:
...
+- `false`: RequireAndVerifyClientCert
```
Comment (dduportal) [inline docs/content/middlewares/forwardauth.md]: Do we have a reference to what `RequireAndVerifyClientCert ` is?

File: docs/content/middlewares/forwardauth.md [hunk 8]
```diff
@@ -266,7 +266,12 @@ http:
...
+- if `tls.ca` is undefined NoClientCert
```
Comment (dduportal) [inline docs/content/middlewares/forwardauth.md]: Do we have a reference to what `NoClientCert` is?

File: docs/content/middlewares/forwardauth.md [hunk 9]
```diff
@@ -454,7 +459,7 @@ http:
 
 #### `tls.insecureSkipVerify`
 
-TODO add description.
+If `insecureSkipVerify` is `true`, TLS for the connection to authentication server accepts any certificate presented by the server and any host name in that certificate.
```
Comment (dduportal) [inline docs/content/middlewares/forwardauth.md]:
```suggestion
Policy for certificate verification for the secured connection to the authentication server.

If `insecureSkipVerify` is `true`, Traefik accepts any certificate presented by the authentication server and any host name in that certificate.
```

File: docs/content/middlewares/forwardauth.md (hunk 5 above)
Comment (dduportal) [reply docs/content/middlewares/forwardauth.md]: Not sure why `Docker` here: assuming this come from copy and paste?

File: docs/content/providers/docker.md [hunk 10]
```diff
@@ -430,7 +430,7 @@ _Optional_
 
 #### `tls.ca`
 
-TODO add description.
+Certificate Authority used for the connection to the connection to Docker.
```
Comment (dduportal) [inline docs/content/providers/docker.md]:
```suggestion
Certificate Authority used for the secured connection to Docker.
```

File: docs/content/providers/docker.md [hunk 11]
```diff
@@ -450,7 +450,12 @@ providers:
 
 #### `tls.caOptional`
 
-TODO add description.
+The policy followed by the server for TLS Client Authentication.
```
Comment (dduportal) [inline docs/content/providers/docker.md]:
```suggestion
Policy followed for the secured connection with TLS Client Authentication to Docker.
```

File: docs/content/providers/docker.md [hunk 12]
```diff
@@ -450,7 +450,12 @@ providers:
...
+Only used if `tls.ca` is defined.
```
Comment (dduportal) [inline docs/content/providers/docker.md]:
```suggestion
Requires `tls.ca` to be defined.
```

File: docs/content/providers/docker.md [hunk 13]
```diff
@@ -470,7 +475,7 @@ providers:
 
 #### `tls.cert`
 
-TODO add description.
+Public certificate used for the connection to Docker.
```
Comment (dduportal) [inline docs/content/providers/docker.md]:
```suggestion
Public certificate used for the secured connection to Docker.
```

File: docs/content/providers/docker.md [hunk 14]
```diff
@@ -493,7 +498,7 @@ providers:
 
 #### `tls.key`
 
-TODO add description.
+Private certificate used for the connection to Docker.
```
Comment (dduportal) [inline docs/content/providers/docker.md]:
```suggestion
Private certificate used for the secured connection to Docker.
```

File: docs/content/providers/docker.md [hunk 15]
```diff
@@ -516,7 +521,7 @@ providers:
 
 #### `tls.insecureSkipVerify`
 
-TODO add description.
+If `insecureSkipVerify` is `true`, TLS for the connection to Docker accepts any certificate presented by the server and any host name in that certificate.
```
Comment (dduportal) [inline docs/content/providers/docker.md]:
```suggestion
Policy for certificate verification for the secured connection to Docker.

If `insecureSkipVerify` is `true`, Traefik accepts any certificate presented by Docker and any host name in that certificate.
```

File: docs/content/providers/marathon.md [hunk 16]
```diff
@@ -404,7 +404,7 @@ _Optional_
 
 #### `tls.ca`
 
-TODO add description.
+Certificate Authority used for the connection to the connection to Marathon.
```
Comment (dduportal) [inline docs/content/providers/marathon.md]:
```suggestion
Certificate Authority used for the secured connection to Marathon.
```

File: docs/content/providers/marathon.md [hunk 17]
```diff
@@ -424,7 +424,12 @@ providers:
 
 #### `tls.caOptional`
 
-TODO add description.
+The policy followed by the server for TLS Client Authentication.
```
Comment (dduportal) [inline docs/content/providers/marathon.md]:
```suggestion
Policy followed for the secured connection to Marathon with TLS Client Authentication.
```

File: docs/content/providers/marathon.md [hunk 18]
```diff
@@ -424,7 +424,12 @@ providers:
...
+Only used if `tls.ca` is defined.
```
Comment (dduportal) [inline docs/content/providers/marathon.md]:
```suggestion
Requires `tls.ca` to be defined.
```

File: docs/content/providers/marathon.md [hunk 19]
```diff
@@ -444,7 +449,7 @@ providers:
 
 #### `tls.cert`
 
-TODO add description.
+Public certificate used for the connection to Marathon.
```
Comment (dduportal) [inline docs/content/providers/marathon.md]:
```suggestion
Public certificate used for the secured connection to Marathon.
```

File: docs/content/providers/marathon.md [hunk 20]
```diff
@@ -467,7 +472,7 @@ providers:
 
 #### `tls.key`
 
-TODO add description.
+Private certificate used for the connection to Marathon.
```
Comment (dduportal) [inline docs/content/providers/marathon.md]:
```suggestion
Private certificate used for the secured connection to Marathon.
```

File: docs/content/providers/marathon.md [hunk 21]
```diff
@@ -490,7 +495,7 @@ providers:
 
 #### `tls.insecureSkipVerify`
 
-TODO add description.
+If `insecureSkipVerify` is `true`, TLS for the connection to Marathon accepts any certificate presented by the server and any host name in that certificate.
```
Comment (dduportal) [inline docs/content/providers/marathon.md]:
```suggestion
Policy for certificate verification for the secured connection to Marathon.

If `insecureSkipVerify` is `true`, Traefik accepts any certificate presented by Marathon and any host name in that certificate.
```

File: docs/content/middlewares/forwardauth.md (hunk 6 above)
Comment (ldez) [reply docs/content/middlewares/forwardauth.md]: nope

File: docs/content/middlewares/forwardauth.md (hunk 7 above)
Comment (ldez) [reply docs/content/middlewares/forwardauth.md]: nope

File: docs/content/middlewares/forwardauth.md (hunk 9 above)
Comment (ldez) [reply docs/content/middlewares/forwardauth.md]: It's not related to Traefik, but to auth server.

File: docs/content/providers/docker.md (hunk 15 above)
Comment (ldez) [reply docs/content/providers/docker.md]: It's not related to Traefik, but to Docker

File: docs/content/providers/marathon.md (hunk 21 above)
Comment (ldez) [reply docs/content/providers/marathon.md]: It's not related to Traefik, but to Marathon

Comment (dduportal) [reply docs/content/providers/marathon.md]: yes, this is the intent of my suggestion? Sounds like it doesn' t work: let's keep the original sentence then.

**Pull request merged**